applications/RomApplication/costum_strategies/rom_builder_and_solver.h

```diff
@@ -278,9 +278,13 @@ class ROMBuilderAndSolver : public BuilderAndSolver<TSparseSpace, TDenseSpace, T
         TSystemVectorType &rX)
     {
         TSparseSpace::SetToZero(rX);
-        for (const auto &node : rNodes){
-            unsigned int node_aux_id = node.GetValue(AUX_ID);
-            const auto &nodal_rom_basis = node.GetValue(ROM_BASIS);
+        auto nodes_begin = rNodes.begin();

```

Comment:
const

and also should be both firstprivate

Reply:
In line 355 we have a similar case, and have not make const nor firstprivate the ModelPart.ElementsBegin() nor ModelPart.ConditionsBegin() variables

Reply:
so the `const` methods are missing?